web/source/osk/oskView.ts

```diff
@@ -851,68 +850,65 @@ namespace com.keyman.osk {
     /**
      * Applies CSS styling and handling needed to perform a fade animation when
      * hiding the OSK.
-     * 
+     *
      * Note:  currently reflects an effectively-dead code path, though this is
      * likely not intentional.  Other parts of the KMW engine seem to call hideNow()
      * synchronously after each and every part of the engine that calls this function,
      * cancelling the Promise.
-     * 
+     *
      * @returns A Promise denoting either cancellation of the hide (`false`) or
      * completion of the hide & its animation (`true`)
      */
+
     protected useHideAnimation(): Promise<boolean> {
       const os = this._Box.style;
       const _this = this;
-      const promise = new Promise<void>(function(resolve) {
-        os.transition='opacity 0.5s linear 0';
 
-        _this._animatedHideResolver = resolve;
+      return new Promise<boolean>(function(resolve) {
+        const cleanup = function() {
+          // TODO(lowpri): attach event listeners on create and leave them there
+          _this._Box.removeEventListener('transitionend', cleanup, false);
+          _this._Box.removeEventListener('webkitTransitionEnd', cleanup, false);
+          _this._Box.removeEventListener('transitioncancel', cleanup, false);
+          _this._Box.removeEventListener('webkitTransitionCancel', cleanup, false);
+          if(_this._animatedHideTimeout != 0) {
+            window.clearTimeout(_this._animatedHideTimeout);
+          }
+          _this._animatedHideTimeout = 0;
+
+          if(_this._Visible && _this.activationConditionsMet) {
+            // Leave opacity alone and clear transition if another element activated
```

Comment:
```suggestion
            // Set opacity and clear transition if another element activated
```

Reply:
It's kinda 'reset opacity to original value' ... 

```suggestion
            // Reset opacity to original value and clear transition if another element activated
```